Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceStateMachine+State.swift

```diff
@@ -40,9 +43,12 @@ extension ExperienceStateMachine {
                     .endingStep(_, let stepIndex, _, _),
                     .endingExperience(_, let stepIndex, _):
                 return stepIndex
+            case .failing(let targetState, _):
+                return targetState.currentStepIndex
             }
         }
 
+        // swiftlint:disable cyclomatic_complexity
```

Comment:
note the disable. It seemed more maintainable in this case to keep all the transitions consolidated in the one spot still, rather than break apart.